src/odemis/util/comp.py

```diff
@@ -124,23 +124,31 @@ def generate_zlevels(actuator, zrange, zstep):
         raise ZeroDivisionError("The step size 'Zstep' can not be zero")
     if zrange[0] == zrange[1] == 0:
         raise ValueError("'zmax' and 'zmin' can not be both zeros")
+    if "z" not in focuser.axes.keys():
+        raise KeyError("The focus actuator %s does not have z axis", focuser)
```

Comment:
In an Exception, the error message must be a single argument => use `%` instead of `,`.